### ISSUE TITLE
`socket_systemd-journal-remote_disabled` fails for CIS profiles after Anaconda installation

### ISSUE THREAD

**Issue by mildas** [ComplianceAsCode/content#12097]
#### Description of problem:
`socket_systemd-journal-remote_disabled` rule fails for all variants of CIS profile after Anaconda installation.

According to final HTML report, the socket is not masked:

<h4><span class="label label-primary">Test that the property LoadState from the systemd-journal-remote.socket is masked</span> 
        <span class="label label-default">oval:ssg-test_socket_loadstate_is_masked_systemd-journal-remote:tst:1</span> 
        <span class="label label-danger">false</span></h4><h5>Following items have been found on the system:</h5>
Result of item-state comparison | Unit | Property | Value
-- | -- | -- | --
false | systemd-journal-remote.socket | LoadState | loaded

Unfortunately, as it was identified by Anaconda test, I don't have report from remediation phase.

#### SCAP Security Guide Version:
latest master

#### Operating System Version:
RHEL 9

#### Steps to Reproduce:

1. Install hardened system using any CIS profile
2. Scan the system

#### Actual Results:
systemd-journal-remote_disabled fails

#### Expected Results:
systemd-journal-remote_disabled passes

**Comment by jan-cerny** [comment]
The report from the remediation phase reveals that the rule wasn't remediated because it passed during the initial scan because the scanner doesn't found the LoadState property of the systemd-journal-remote.socket, IOW no items have been found conforming to OVAL object oval:ssg-test_socket_loadstate_is_masked_systemd-journal-remote:tst:1. It isn't clear whether the LoadState property doesn't exist or the systemd-journal-remote.socket doesn't exist. 

Does https://bugzilla.redhat.com/show_bug.cgi?id=1999587 strike back?

**Comment by jan-cerny** [comment]
I have performed an experiment.

I have run a manual installation of a virtual machine of a released RHEL 9.4. I have used a modified OAA which I have modified to produce ARF instead XCCDF results in /root/openscap_data. I have used SCAP source data stream built locally from current upstream master as of 2024-06-08 as of HEAD 7cde9d18a5e1441039a3c0d0e5bcc1a77efe9880. I have used CIS server L2 profile.  I have configured partitioning according to products/rhel9/kickstart/ssg-rhel9-cis-ks.cfg. The installation has been completed successfully.

Before the reboot, I have verified in shell in TTY1 that systemd-journal-remote.socket doesn't exist in the installer environment.

After the reboot, I have seen that systemd-journal-remote.socket isn't masked, and therefore this issue has been successfully reproduced in my experiment.

On the produced ARF and report from the remediation phase I have verified that the rule systemd-journal-remote_disabled passes during the installation because the  systemd-journal-remote.socket hasn't been found. Therefore no remediation has been executed for the rule systemd-journal-remote_disabled.

Based on this experiment, it's obvious that this issue is caused by https://bugzilla.redhat.com/show_bug.cgi?id=1999587.

In other words, the problem is that OpenSCAP running in the Anaconda environment gets data about systemd units from the installer environment, not from the installed system. Specifically, in this situation, systemd-journal-remote.socket  exists in the installed system, but OpenSCAP doesn't find this unit, because it gets the data about systemd units from the installer environment where this unit doesn't exist. Consequently, the rule passes incorrectly and isn't remediated.

**Comment by jan-cerny** [comment]
This is an issue in OAA and we have reported it to downstream in https://issues.redhat.com/browse/RHEL-45706

**Comment by mildas** [comment]
As this issue has been reported to RHEL Jira, closing this one to not have it reported on two places.
